### PR TITLE
Add Spring Boot example of usage

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,21 +38,9 @@
 
         <!-- Test dependencies -->
         <dependency>
-            <groupId>org.slf4j</groupId>
```
Comment (bsideup) [inline pom.xml:40]: I had to remove it from here because there were some conflicts with Spring Boot

**Comment by bsideup** [comment]
@rnorth added some simple example, please take a look once you have a time :)

**Comment by bsideup** [comment]
@rnorth any news?

**Comment by rnorth** [comment]
@bsideup sorry, I thought I'd already processed this - apologies!

File: spring-boot/src/test/java/com/example/AbstractIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @Configuration
+    @Import(DemoApplication.class)
+    static class TestConfiguration {
+
+        @Autowired
+        ConfigurableEnvironment environment;
+
+        @PostConstruct
+        public void init() {
+            environment.getPropertySources().addFirst(new PropertySource("containers") {
```
Comment (rnorth) [inline spring-boot/src/test/java/com/example/AbstractIntegrationTest.java]: Could this not be simplified through use of a MockPropertySource object?

File: spring-boot/pom.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+	<parent>
+		<artifactId>testcontainers-java-examples</artifactId>
```
Comment (rnorth) [inline spring-boot/pom.xml]: Sorry to bring it up but please could you use spaces, not tabs, for indentation..?

Comment (bsideup) [inline spring-boot/pom.xml]: yes, sure, missed that somehow :)

File: spring-boot/src/test/java/com/example/AbstractIntegrationTest.java (hunk 2 above)
Comment (bsideup) [inline spring-boot/src/test/java/com/example/AbstractIntegrationTest.java]: Initially I thought that "on demand" model will be better, but actually now I think you're right and pre-set key/value pairs will be good enough for it

**Comment by bsideup** [comment]
@rnorth I'm sorry, somehow I forgot about this PR :$ applied all fixes, please take a look :)

**Comment by bsideup** [comment]
@rnorth ping :) CircleCI seems to fail on "checkDiskSpace" - should not be related to the example

**Comment by bsideup** [comment]
@rnorth ping

**Comment by bsideup** [comment]
hey @rnorth !

I found a way how to make Spring Boot example more reliable and simple, could you please take a look? Thanks!

**Comment by rnorth** [comment]
Ah cool! I'll merge now - there is a test failure but it's unrelated, and I don't want to hold this up any longer.

**Comment by bsideup** [comment]
@rnorth awesome, thanks!

**Pull request merged**